src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -312,6 +312,17 @@ void remove_redundant_reorders::run(program& p) {
         if (!o_layout.compatible(i_layout))
             continue;
 
+        // Check peer node which is reorder with padding. The reorder will cause layout incompatible.
+        // So r_node shouldn't be opt out in this case.
+        auto& dep = r_node.get_dependency(0);
+        auto will_dependency_have_padding = false;
+        for (auto user : dep.get_users()) {
+            if ((user != node) && (user->is_type<reorder>()) && (user->get_output_layout().data_padding))
+                will_dependency_have_padding = true;
+        }
+        if (will_dependency_have_padding == true)
+            continue;
```

Comment:
Isn't it okay just to mark it can_be_optimized?

Reply:
Also this naming is ambiguous, the "dependency" does not have padding but the "other user of dep" has padding, shown in the condition.

Reply:
Updated fix code as discussed. Thanks!